### PR TITLE
Unify dotnet/runtime on using Arcade -publish

### LINKED ISSUE #4239: Unify publishing infrastructure in runtime
See discussion in https://github.com/dotnet/source-build/issues/4101#issuecomment-2000216765

Runtime is the remaining repository that publishes differently when building the repo in isolation versus when building inside the VMR. The VMR publishing is also missing some artifacts, i.e. checksums, version files and potentially more.

This item tracks getting rid of `prepare-artifacts.proj` in runtime and using the Arcade publish entry-point. Presumably this just means following the already established process from windowsdesktop and deployment-tools: https://github.com/dotnet/windowsdesktop/pull/4251

### PR DESCRIPTION
**Pull request by jkoritzinsky**
Fixes https://github.com/dotnet/source-build/issues/4239

Replaces https://github.com/dotnet/runtime/pull/100004

Current official build link: https://dev.azure.com/dnceng/internal/_build/results?buildId=2566438&view=results

### CODE REVIEW

File: eng/Publishing.props [hunk 1]
```diff
@@ -1,39 +1,194 @@
...
+    Look through the downloaded artifacts to figure out how to ship them. Creates item groups for
+    other types of publishing to use.
+
+    DownloadDirectory is expected to have some directory inside, which then contains a dir for each
+    build job name. For example, this nupkg would be found:
+
+      $(DownloadDirectory)IntermediateArtifacts\windows_x64\Shipping\a.nupkg
+  -->
+  <ItemGroup Condition="'$(DotNetBuild)' != 'true'">
+    <Error Text="DownloadDirectory is not set." Condition="'$(DownloadDirectory)' == ''" />
```
Comment (ViktorHofer) [inline eng/Publishing.props:21]: Move this line into the `ValidateDownloadedAssets` target?

**Comment by ViktorHofer** [comment]
Unfortunately this won't work because of https://github.com/dotnet/arcade/blob/31624193093a13f765ab5382509e693911264509/src/Microsoft.DotNet.Arcade.Sdk/tools/Build.proj#L98

You will notice this when you update this branch with latest main (which has an updated Arcade). You can fix this by adding an "Empty.proj" file which as the name says is empty and pass that in as a `ProjectToBuild` item.

**Comment by jkoritzinsky** [comment]
I have an idea on how to avoid the empty.proj pattern. I'll try it out and see what happens.

**Comment by ViktorHofer** [comment]
Yeah that's more or less the same just without an additional project. You are now also building the repo tasks unnecessarily but that should be fine.

**Comment by ViktorHofer** [comment]
> D:\a\_work\1\s\.packages\microsoft.dotnet.arcade.sdk\10.0.0-beta.24511.1\tools\Publish.proj(310,5): error MSB4044: The "PublishSymbols" task was not given a value for the required parameter "PersonalAccessToken".

**Review by ViktorHofer — APPROVED** [review]

**Comment by jkoritzinsky** [comment]
Looks like we have some repeated assets from the productVersion.txt files. I'll figure out a solution tomorrow.

**Review by ViktorHofer — APPROVED** [review]

**Comment by ViktorHofer** [comment]
The official build succeeded. Merging to unblock https://github.com/dotnet/sdk/pull/44014.

We can follow-up on whether the main branch should sign everything (real) or not in a follow-up.

**Pull request merged**